src/core/expression/qgsexpressionfunction.cpp

```diff
@@ -5545,6 +5545,25 @@ static QVariant fcnHashSha256( const QVariantList &values, const QgsExpressionCo
   return fcnHash( QgsExpressionUtils::getStringValue( values.at( 0 ), parent ), QCryptographicHash::Sha256 );
 }
 
+static QVariant fcnToBase64( const QVariantList &values, const QgsExpressionContext *, QgsExpression *parent, const QgsExpressionNodeFunction * )
+{
+  QString value = QgsExpressionUtils::getStringValue( values.at( 0 ), parent );
+
+  const QByteArray input = value.toLocal8Bit();
```

Comment:
I think this should never go through a `QString` or a `toLocal8Bit` and be passed as raw bytes to tobase64.

Reply:
Right - this should be:

    const QByteArray input = values.at( 0 ).toByteArray();
    return QVariant( QString( input.toBase64() ) );